packages/common/src/core/behaviors/IWithEditorBehavior.ts

```diff
@@ -0,0 +1,88 @@
+import { JSONSchema } from "json-schema-typed";
+import Ajv from "ajv";
+
+export enum UiSchemaRuleEffects {
+  SHOW = "SHOW",
+  HIDE = "HIDE",
+  DISABLE = "DISABLE",
+  NONE = "",
+}
+
+export enum UiSchemaElementTypes {
+  section = "Section",
+  step = "Step",
+  control = "Control",
+  layout = "Layout",
+  slot = "Slot",
+}
+
+export enum UiSchemaSectionTypes {
+  accordion = "accordion",
+  stepper = "stepper",
+}
+export interface IConfigurationSchema extends JSONSchema {
+  type?: "object";
+}
+
+export interface IConfigurationValues {
+  [key: string]: unknown;
+}
+
+export interface IChangeEventDetail {
+  valid: boolean;
+  values?: {
+    [key: string]: any;
+  };
+}
+
+export interface IValidationResult {
+  valid: boolean;
+  errors?: Ajv.ErrorObject[];
+}
+
+export interface IUiSchemaRule {
+  effect: UiSchemaRuleEffects;
+  condition: {
+    scope: string;
+    schema: IConfigurationSchema;
+  };
+}
+
+export interface IUiSchemaElement {
+  type: string;
+  labelKey?: string;
+  label?: string;
+  options?: {
+    [key: string]: any;
+  };
+  scope?: string;
+  rule?: IUiSchemaRule;
+  elements?: IUiSchemaElement[];
+  tooltip?: string;
+}
+export interface IUiSchema extends IUiSchemaElement {
+  elements?: IUiSchemaElement[];
+}
+
+export type EditorConfigType = "complete" | "minimal";
```

Comment:
I don't have super strong feelings about this, but just wondering why we decided to go with "minimal" | "complete" vs. the more explicit "create" | "edit" types?

Technically the "create" form will be more "minimal" in that it will have less fields than the complete "edit" form, but I feel like it might also be a bit confusing to just call the "create" form "minimal" because there's also an implicit layout (i.e. stepped) that's associated with it. Not sure if this makes sense, but just a thought.

Reply:
Love it. Many times when I'm 🤔  about naming etc, I'll just slap in something and worry about better names later - this is exactly one of those cases. I'll change to `create` and `edit` :)